### ISSUE TITLE
adapt RIOTs development procedures

### ISSUE THREAD

**Issue by mfrey** [cn-uofbasel/ccn-lite#221]
In order to engage more fellow hackers to contribute to CCN-lite, I would propose that we first adapt [RIOTs development procedures](https://github.com/RIOT-OS/RIOT/wiki/Development-procedures) for CCN-lite and second, have a look at [RIOTs community Processes](https://github.com/RIOT-OS/RIOT/wiki/RIOT-Community-Processes).

We don't have to use RIOTs defined guides and processes, but I think it is a good starting point. I suggest that I will set up the corresponding wiki pages and we can discuss the pages here (in this issue). If it comes down to mailing lists/f2f meetings etc. I think this requires a "critical mass" and can be handled later. 

I think we have some bits & pieces already in place, but we can improve. Like the [linux kernel janitor project](https://kernelnewbies.org/KernelJanitors) we could for example think of introducing janitor tasks, i.e. simple contributions tasks for people new to CCN-lite (the basis could be our TODO list in the wiki). 

Any thoughts? Particularly, @cgundogan @blacksheeep @cmarxer @waehlisch

**Comment by blacksheeep** [comment]
I support that. And I am looking forward to your wiki page. 
The janitor task sounds interesting, depending on how many people using this project are willing to contribute.

**Comment by cgundogan** [comment]
Yes, I also agree on this. We should definitely use RIOTs community processes as a basis to come up with some simple rules and practices. Another important step for community bulding is to finally activate the mailing list. @blacksheeep do you have access to ccn-lite.net and can you provide a mailing list on that domain?

**Comment by blacksheeep** [comment]
The domain is managed by ITS-Unibas. As far as I know, I will not get a mailing list on that domain, since they have to be ending with unibas.ch by policy. 
We already have: 
ccn-lite@maillist.unibas.ch

**Comment by cgundogan** [comment]
@blacksheeep oh okay. How can I subscribe there?

**Comment by blacksheeep** [comment]
Here you go:
https://www.maillist.unibas.ch/mailman/listinfo/ccn-lite

**Comment by mfrey** [comment]
I've clearly underestimated the work load. This is clearly going to take some time. I've put up a preliminary (obviously)  [development procedures](https://github.com/cn-uofbasel/ccn-lite/wiki/Development-procedures) which in turn point to non-existing or half-filled/empty wiki pages on other topics.

**Comment by mfrey** [comment]
I've also added the [community process](https://github.com/cn-uofbasel/ccn-lite/wiki/Community-processes). If you have some time, please go over the two pages and make changes or propose changes over here. 

I'm not entirely sure about the maintainer/contributor thing. I'm a contributor, but not a maintainer (if it comes down to the definition), but have PR rights - we probably should align this.

Please also check the [maintainers list](https://github.com/cn-uofbasel/ccn-lite/wiki/Maintainers) in the wiki.

**Comment by blacksheeep** [comment]
That looks all pretty good!

**Comment by PeterKietzmann** [comment]
@mfrey the wiki looks promising! In my personal opinion pages such as "maintainers" and "roadmap" tend to become outdated. Furthermore, "roadmaps" tend to duplicate issues. How about pointing to a public peoples/maintainers list provided by GitHub (something like [https://github.com/orgs/RIOT-OS/people](https://github.com/orgs/RIOT-OS/people) ) and to a "search by label" in the issue tracker for the roadmap?

**Comment by mfrey** [comment]
@PeterKietzmann Thanks for the feedback! I totally agree, we should avoid to provide information which becomes soon outdated. The roadmap page was named todo before and probably covered the purpose better - it was and probably still is a list of things various people noticed while going through the code.

About the people/maintainers list - as far as I can see RIOT has different roles which are provided by GitHubs organization mechanisms. In order to copy this "behaviour" people would need to become members of the cn-uofbasel group or ccn-lite moves to a new group maintained by cn-uofbasel and a few other (active) members/contributors of ccn-lite, saying that I'm not really sure if the project is ready for this huge step. But, maybe you meant something completely different?
